plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2488,7 +2488,17 @@ protected ConsoleDef createConsoleDef() {
         return new ConsoleDef(PTY, null, null, (short)0);
     }
 
-    protected VideoDef createVideoDef() {
+    protected VideoDef createVideoDef(VirtualMachineTO vmTO) {
+        Map<String, String> details = vmTO.getDetails();
+        if (details != null) {
+            if (details.containsKey(VideoDef.VIDEO_MODEL)) {
+                _videoHw = details.get(VideoDef.VIDEO_MODEL);
```

Comment:
@leolleeooleo 
_videoHw and _videoRam are global variables. I think local variables should be used here.

Reply:
I copy from https://github.com/apache/cloudstack/blob/789d68e43b9e70eecdebbcbf09fcacf25212cc63/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java#L1040

The value _videoHw and _videoRam are load from /etc/cloudstack/agent/agent.properties
If details value exist will override the value.

Do you mean:
```
    protected VideoDef createVideoDef(VirtualMachineTO vmTO) {
        Map<String, String> details = vmTO.getDetails();
        if (details != null) {
            if (details.containsKey(VideoDef.VIDEO_MODEL)) {
                String videoHw = details.get(VideoDef.VIDEO_MODEL);
                int videoRam = 16384;
                if (details.containsKey(VideoDef.VIDEO_RAM)) {
                    String value = details.get(VideoDef.VIDEO_RAM);
                    videoRam = NumbersUtil.parseInt(value, videoRam);
                }
            return new VideoDef(videoHw, videoRam);
            }
        }
        return new VideoDef(_videoHw, _videoRam);
    }
```

Reply:
@leolleeooleo exactly. we should not change the global variables.

you can use 
```
String videoHw = _videoHw;
int videoRam = _videoRam;
if (details != null) {
      ...... // update videoHw and videoRam
}
return new VideoDef(videoHw, videoRam);
```